### PR TITLE
Bug fix analysis stability

### PR DESCRIPTION
**Pull request by DSWallach**
# Pull Request Template for MMEDS

## What has changed
Include each file that was changed, how it was changed (in broad terms), and why it was changed.
host/host.py: The Watcher object is now created outside the server and passed in as a parameter.
As I discovered, due to the different sub-classes in the server there were five separate watchers created with five separate queues. I think this was the cause of the issue running analysis during the demo.
mmeds/database.py: The MetadataUploader class now inherits Process, and acts like one. This is to allow calling `get_info` to write to the process log.
mmeds/server.py: Changes to how the Watcher is created and shared.
mmeds/spawn.py: Enhancements to the Watcher class for improved monitoring of running processes
## Checklist of pre-requisites
-   [x] Does the code run?  
-   [x] Does the code follow the repository style?  
-   [x] Is the code tested?  

## How to use the feature
Example: To do X run new function Y with parameters A, B, and C which gives result Z.

## How the design has changed
Example: Functions A, B, and C are now all methods of Class D.

### CODE REVIEW

File: mmeds/config.py [hunk 1]
```diff
@@ -37,7 +37,10 @@
 MMEDS_LOG = DATABASE_DIR / 'mmeds_log.txt'
 SQL_LOG = DATABASE_DIR / 'sql_log.txt'
 DOCUMENT_LOG = DATABASE_DIR / 'document_log.txt'
-PROCESS_LOG = DATABASE_DIR / 'process_log.txt'
+PROCESS_LOG_DIR = DATABASE_DIR / 'process_log_dir'
+if not os.path.exists(PROCESS_LOG_DIR):
```
Comment (mstapylton) [inline mmeds/config.py]: not pathlib?!

File: mmeds/database.py [hunk 2]
```diff
@@ -766,17 +767,47 @@ def __init__(self, metadata, path, owner, study_type, reads_type, study_name, te
...
+            self.access_code = access_code
+
+        count = 0
+        new_dir = fig.DATABASE_DIR / ('{}_{}_{}'.format(self.owner, self.study_name, count))
+        while new_dir.is_dir():
+            new_dir = fig.DATABASE_DIR / ('{}_{}_{}'.format(self.owner, self.study_name, count))
+            count += 1
+        new_dir.mkdir()
+
+        self.path = Path(new_dir) / 'database_files'
```
Comment (mstapylton) [inline mmeds/database.py]: does this work for concatenating the paths? I've been using Path.joinpath()

File: mmeds/spawn.py [hunk 3]
```diff
@@ -162,28 +106,98 @@ def __init__(self, queue, parent_pid, testing=False):
...
+        Part of the functionality for continuing unfinished analyses on server restart.
+        """
+        if fig.CURRENT_PROCESSES.exists():
+            with open(fig.CURRENT_PROCESSES, 'r') as f:
+                processes = yaml.safe_load(f, Loader=yaml.Loader)
+            for key in processes.keys():
+                for process in processes[key]:
+                    process['is_alive'] = False
+        else:
+            processes = defaultdict(list)
```
Comment (mstapylton) [inline mmeds/spawn.py]: I think this is a good example of a situation where I might consider adding logging like:
logger.warn("Current_Processes not found, returning defaultdict instead")
Not sure if that's needful here, but I like keeping an eye out for places where an unexpected code path can cause issues further downstream.

File: deblur.log [hunk 4]
```diff
@@ -0,0 +1,11 @@
+INFO(140629163607872)2019-10-21 09:49:33,995:*************************
```
Comment (mstapylton) [inline deblur.log]: A log file should probably not be in the PR. Is there any way you could move it into its own folder and then add that to the .gitignore?

File: mmeds/config.py (hunk 1 above)
Comment (DSWallach) [reply mmeds/config.py]: That is shameful. I just copied how I set up other directories back when I made this file at the beginning of mmeds

File: mmeds/database.py (hunk 2 above)
Comment (DSWallach) [reply mmeds/database.py]: Yeah, this is the best part of pathlib. It'll automatically handle the right delimiters '\' for windows, fixing double '//'

File: mmeds/spawn.py (hunk 3 above)
Comment (DSWallach) [reply mmeds/spawn.py]: I should remove this whole function. I have decided against loading general processes from a log. To continue tracking jobs when restarting on Minerva I think I will add something else, like a function that runs 'bjobs' and grabs the output or something

File: deblur.log (hunk 4 above)
Comment (DSWallach) [reply deblur.log]: This log file has been the bane of my existence. It will get re-created there the next time deblur is run even if I move it.

Comment (mstapylton) [reply deblur.log]: That is annoying, well it's a small issue and I trust you've tried the easy ways of remedying it.

File: mmeds/spawn.py (hunk 3 above)
Comment (mstapylton) [reply mmeds/spawn.py]: Makes sense, you could try setting up checkpointing and automating running jobs from a checkpoint.

Comment (DSWallach) [reply mmeds/spawn.py]: That is something I should have setup for larger datasets in the future but would address the issue of continuing to monitor an analysis after a mmeds server restart

**Pull request merged**